### PR TITLE
feat: more sophisticated method to get copilot path

### PR DESCRIPTION
**Pull request by rockofox**
Don't rely on a fixed `plugin_manager_path` to find the plugin base path. Not only does that require changing an option depending on what plugin manager the user uses, but it also makes it incompatible with some plugin managers (i.e Nix, which doesn't store plugins under their literal name).

This PR uses [`debug.getinfo(1).source:sub(2)`](https://neovim.discourse.group/t/get-path-to-plugin-directory/2658) to find the current lua file and traverses to the base directory from there. This is much more versatile and portable.

### CODE REVIEW

**Comment by zbirenbaum** [comment]
I think this was proposed a while back, but I wasn't a huge fan of using debug functions as part of the plugin and tried to find another way. Unfortunately the way I came up with broke under a variety of circumstances so it was reverted.

Since debug is probably the only way of doing this, I guess it might make sense. What do you think @MunifTanjim?

**Comment by MunifTanjim** [comment]
Was `vim.api.nvim_get_runtime_file('plugin/copilot.lua', false)` considered before?

**Comment by zbirenbaum** [comment]
> vim.api.nvim_get_runtime_file('copilot/index.js', false)

I can't remember exactly  but this seems to work great. I'll request it as a change.

File: lua/copilot/util.lua [hunk 1]
```diff
@@ -229,12 +229,12 @@ function M.get_editor_configuration()
...
 end
 
-M.get_copilot_path = function(plugin_path)
-  for _, loc in ipairs({ "/opt", "/start", "" }) do
-    local copilot_path = plugin_path .. loc .. "/copilot.lua/copilot/index.js"
-    if vim.fn.filereadable(copilot_path) ~= 0 then
-      return copilot_path
-    end
+M.get_copilot_path = function()
+  local copilot_path = string.gsub(debug.getinfo(1).source:sub(2), "lua/copilot/util.lua", "") .. "copilot/index.js"
```
Comment (zbirenbaum) [inline lua/copilot/util.lua:233]: change to using `vim.api.nvim_get_runtime_file('copilot/index.js', false)`

**Pull request merged**